gordo/machine/validators.py

```diff
@@ -133,9 +133,22 @@ class ValidMachineRuntime(BaseDescriptor):
     def __set__(self, instance, value):
         if not isinstance(value, dict):
             raise ValueError(f"Runtime must be an instance of dict")
+        self._verify_reporters(value)
         value = fix_runtime(value)
         instance.__dict__[self.name] = value
 
+    def _verify_reporters(self, value: dict):
+        """
+        Verify the expected existence and structure of runtime.reporters
+        """
+        if "reporters" not in value:
+            value["reporters"] = list()
```

Comment:
It seems off to modify the dict during validation. Could the runtime be a dataclass that would then enforce the empty list if one is not passed when the runtime "data" is initialized? That way there would be a one-stop-shop for looking at what are the requisite properties of `runtime` while avoiding an ambiguous injection of the field when missing. 

This of course could/should be another PR if it means some bigger overhauling.

Reply:
I don't think that because the dict is missing a key makes it invalid, but I agree, the runtime might be nice to have as a dataclass, and would address this anyhow, but should be addressed outside of this PR IMO.